common/views/components/ButtonSolidLink/ButtonSolidLink.tsx

```diff
@@ -12,9 +12,18 @@ import ConditionalWrapper from '../ConditionalWrapper/ConditionalWrapper';
 
 type ButtonSolidLinkProps = ButtonSolidBaseProps & {
   clickHandler?: (event: SyntheticEvent<HTMLAnchorElement>) => void;
-  link: {href: {pathname: string, query: string}, as: {pathname: string, query: string}} | string;
+  link:
+    | {
+        href: { pathname: string; query: string };
+        as: { pathname: string; query: string };
+      }
+    | string;
 };
 
+export function getHref(link: string | object): string | undefined {
```

Comment:
Could we have a function that returns a type that this component expects?

Reply:
yeah, I'm learning about overloads and trying to rewrite this in a way that does that.

Reply:
I thought overloads might be the answer, but maybe not. [This is what I expected to work](https://www.typescriptlang.org/play?#code/C4TwDgpgBAchAewAyBLAdgawCrmgXigG8AoKKACwCcIAzALiKjAENhy1mBbCBgZ2EroA5gG4oARwCuESiD4DhUAL4jSUZrwaEmrdlx5R+gtKInTZ840OWqVxYjUloAxsBQB7NFCERgACWoaAAoAG3QMBjhEVEwcSABKBicAE1p0CGTVRxc3T29fANpQ8MthRMMFEyynVw8vH39A4sxSkygAH1gEZHC4iHKjRU6UtLQMojVqYElKL1BIdxooMMwoPHWoAHJ3ACMAKwhXTagAfigRmnTkqAYVjFt7Z09+KDHokq732Nw1oipaLQsNgcbgMTY7ELMcibAA0Zhkci2EPc7jAmyUcI0gN0IIM4Mh0LhUgRYORqPRdieaBeEKhrWsBHxUM2qipL0GJgA8pQoj1VgQgvE1gA+CZkKYzLwAWV0ADpKMw0MlBVBRQAGWUAVlOr26MQwNygtPID2IDUKwWN8VU5qabz5GGtZoKTQ5Qm5vP1gutQA) because I thought the overloads would be able to distinguish between the `link` prop which is `string | NextLinkType`, but doesn't look like it.

Reply:
This feels like a bit dance to please TS – is there a simpler way I'm missing?